src/resources/telemetry/index.ts

```diff
@@ -3,3 +3,71 @@
 export { Logs, type LogListResponse, type LogListParams } from './logs';
 export { Telemetry } from './telemetry';
 export { Traces, type TraceListResponse, type TraceListParams } from './traces';
+import type * as Core from '../../core';
+import type * as Shared from '../shared';
+
+export interface BaseListParams {
+  limit: number;
+  start?: string;
+  end?: string;
+}
+
+export interface StreamableResource<TData, TParams extends BaseListParams> {
+  list(
+    query: TParams,
+    options?: Core.RequestOptions,
+  ): Core.APIPromise<{
+    data: Array<TData>;
+    links: Shared.PaginationLinks;
+  }>;
+}
+
+export interface StreamParams {
+  frequency?: number;
+}
+
+export async function* streamTelemetryData<TData, TParams extends BaseListParams>(
+  resource: StreamableResource<TData, TParams>,
+  query: StreamParams,
+  options?: Core.RequestOptions,
+): AsyncGenerator<TData> {
+  const { frequency = 1000 } = query;
+
+  while (true) {
+    let startTime = new Date(Date.now() - frequency);
+    let endTime = new Date();
+
+    // Gather the range startTime:=(now - frequency) --> now
+    let prevNextLink = '';
+    while (true) {
+      const listParams = {
+        limit: 20,
+        start: startTime.toISOString(),
+        end: endTime.toISOString(),
+        ...query,
+      } as TParams;
```

Comment:
Somewhat nervous that this doesn't actually do type checking in the hard way - we can't say `satisfies TParams` here or `const listParams: TParams = ` here because concrete instances can't have generic types. Probably not a big thing but I don't think TypeScript is protecting us that much here.